packages/ibm-products/src/components/StringFormatter/StringFormatter.js

```diff
@@ -0,0 +1,176 @@
+/**
+ * Copyright IBM Corp. 2024, 2024
+ *
+ * This source code is licensed under the Apache-2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+// Import portions of React that are needed.
+import React from 'react';
+
+// Other standard imports.
+import PropTypes from 'prop-types';
+import cx from 'classnames';
+
+import { getDevtoolsProps } from '../../global/js/utils/devtools';
+import { pkg /*, carbon */ } from '../../settings';
+
+// Carbon and package components we use.
+import { DefinitionTooltip } from '@carbon/react';
+import { StringFormatterAlignment } from './utils/enums';
+// The block part of our conventional BEM class names (blockClass__E--M).
+const blockClass = `${pkg.prefix}--string-formatter`;
+const componentName = 'StringFormatter';
+
+// NOTE: the component SCSS is not imported here: it is rolled up separately.
+
+// Default values for props
+const defaults = {
+  lines: 1,
+  tooltipDirection: StringFormatterAlignment.BOTTOM_LEFT,
+  truncate: false,
+  width: null,
+};
+
+/**
+ * StringFormatter allows for truncating text while displaying a tooltip
+ * overlay on hover or focus with the entirety of the provided copy.
+ */
+export let StringFormatter = React.forwardRef(
+  (
+    {
+      // The component props, in alphabetical order (for consistency).
+
+      className,
+      lines = defaults.lines,
+      tooltipDirection = defaults.tooltipDirection,
+      truncate = defaults.truncate,
+      width = defaults.width,
+      value,
+      /* TODO: add other props for StringFormatter, with default values if needed */
+
+      // Collect any other property values passed in.
+      ...rest
+    },
+    ref
+  ) => {
+    return (
+      <span
+        className={cx(
+          blockClass, // Apply the block class to the main HTML element
+          className // Apply any supplied class names to the main HTML element.
+        )}
+        ref={ref}
+        {
+          // Pass through any other property values as HTML attributes.
+          ...rest
```

Comment:
prop spreading should always be first in order to avoid the user possibly overwriting them